substrate/client/service/src/lib.rs

```diff
@@ -451,7 +452,7 @@ where
 	match tokio::task::block_in_place(|| {
 		tokio_handle.block_on(sc_rpc_server::start_server(server_config))
 	}) {
-		Ok(server) => Ok(Box::new(waiting::Server(Some(server)))),
+		Ok(server) => Ok(server),
```

Comment:
use waiting::Server(server) here
```suggestion
		Ok(server) => Ok(waiting::Server(server)),
```